functions/user-defined-literals.md

```diff
@@ -0,0 +1,112 @@
+# Functions: user-defined literals
+_Skeleton instructions are typeset in italic text._
+
+## Overview
+
+_Provides a short natural language abstract of the module’s contents._
+_Specifies the different levels of teaching._
+
+<table>
+  <thead>
+    <th>Level</th>
+    <th>Objectives</th>
+  </thead>
+  <tr>
+    <td>Foundational knowledge</td>
+    <td>using and understanding UDLs</td>
+  </tr>
+  <tr>
+    <td>Advanced</td>
+    <td>implementing your own UDLs</td>
+  </tr>
+  <tr>
+    <td>Further studies</td>
+    <td>Advanced use (<code>"{}, {}!"_fmt("Hello", "World")</code>)</td>
+  </tr>
+</table>
+
+## Motivation
+
+_Why is this important?_
+_Why do we want to learn/teach this topic?_
+
+* Allows clearer expression of intent in C++.
+* `std::string`: `"Hello, world!"s`
+* `std::chrono`: `3h + 10m + 5s`
+
+## Topic introduction
+
+_Very brief introduction to the topic._
+
+* Explain the existence of user defined literals. Example: `12m + 17s` is terse,
+  expressive and type safe.
+
+## Foundational knowledge: Using UDLs
+
+### Background/Required Knowledge
+
+A student:
+* knows how to form numeric literals e.g. `1.5f` means a `float` of value `1.5`.
+* is familiar with the major C++ types:
```

Comment:
```suggestion
* is familiar with the major C++ types [[C++ object model: types]](https://github.com/cplusplus/SG20/):
```
we should add cross references to other modules/topics.
I'm currently not sure how to best but this into md.
Options:
1. after text [Module: Topic] // (suggested above)
2. [Module: Topic] before text
3. after text right aligned:                                           [Module: Topic]

I would prefer option 3. but somehow the github markdown does align <a> well and if I wrap it into a <div> it adds a newline :cry: , do you know some md magic to do this?

Reply:
Please use reference links instead of using inline links.

https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#links

Reply:
I updated the link.
If you think this is fine otherwise, I'll update the rest of the references.

Reply:
@cjdb I update the link, according to what we discussed.
I think, I also found a missing topic. Do we currently have a `namespaces` topic?
Maybe I missed it but I could not find it in p1725.